File: solutions/482.license-key-formatting/license-key-formatting.js
/**
 * @param {string} S
 * @param {number} K
 * @return {string}
 */
var licenseKeyFormatting = function(S, K) { 
    S = S.split('-').join('').toUpperCase();
    const len = S.length;
    if(len === 0) return '';
    let start = 0;
    let res = '';
    if((len % K) === 0){
        res += S.substring(start, start + K);
        start += K;
    }else{
        res += S.substring(start, start + (len % K));
        start += (len % K);
    }
    for(let i = start; i < len; i += K){
        res += '-' + S.substring(i, i + K);
    }
    return res;
};

File: solutions/482.license-key-formatting/license-key-formatting.java
class Solution {
    public String licenseKeyFormatting(String S, int K) {
        StringBuilder res = new StringBuilder();
        int cnt = 0;
        S = S.toUpperCase();
        for(int i = 0; i < S.length(); i++){
            char c = S.charAt(S.length() - 1 - i);
            if(c == '-') continue;
            
            if(cnt != 0 && cnt % K == 0){
                res.append('-');
            }
            res.append(c);
            cnt++;
        }
        return res.reverse().toString();
        
    }
}
